Add project details to the README file (#237)
google/android-fhir: README.md

# Android FHIR SDK (early access) ![master](https://github.com/google/android-fhir/workflows/CI/badge.svg?branch=master)

The Android FHIR SDK is an Android library that enables developers to develop offline, mobile first healthcare applications that adhere to the FHIR specification. The SDK is being designed to support a number of common mobile health use cases including:

- FHIR Clinical Data Repository
- FHIR native data capture for case reporting
- Client tracking and care coordination applications
- Native support for on-device clinical decision support (via use of CQL)
- Ability to run certain FHIR operations for generation of Care Plans and Indicators (for example)

The overall goal is to simplify the process of incorporating support for FHIR into new or existing mobile solutions and to accelerate the adoption of FHIR standards as part of broader interoperability efforts in Healthcare.

**The Android SDK is a community collaboration and contributions to the codebase, feature requests and suggestions about specific use cases are welcomed.** 

> If you want to provide any feedback, discuss specific use cases or have questions about getting involved, please email us at android-fhir-sdk@google.com

Author: I've re-organised based on Julies feedback. PTAL Thanks!

For developers looking to get started with the SDK, see the section on Contributing

> ## Status

Reviewer: Can we move this a bit more up on the page? It's a critical piece of info and currently feels a bit lost in the middle of the content

Author: Done.

**This is currently in Early Access for Developers ONLY and is NOT production-ready. Do NOT use in production.**

> ## Usage and Features

Reviewer: Can we remove the list of features from the readme? I feel this takes us a bit too far from the current [readme recommendations](https://github.com/18F/open-source-guide/blob/18f-pages/pages/making-readmes-readable.md) and would prefer that we keep the content to [core below points](https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/about-readmes#section-links-in-readme-files-and-blob-pages):
- What the project does
- Why the project is useful
- How users can get started with the project
- Where users can get help with your project
- Who maintains and contributes to the project

The SDK is designed to support Android 21 (lollipop) and above
 
The FHIR SDK will provide a rich set of features for building FHIR native solutions including:

- Data Access API (FHIR resource management using FHIR protos)
- Storage API (via Sqlite DB)
- Sync API (sync capabilities to a FHIR server)
- Search API
- Data Capture API (Structured Data Capture library)
- CQL Engine (using https://github.com/DBCG/cql_engine)

## Details of the repository
The repository is organised into two main libraries, *core* and *structured data capture* as well as sample demo applications for each (see sample applications).

### Core library:
This is the main SDK library providing core functionality required for use cases that require a mobile based clinical data repository using FHIR. This is currently in active development with a planned alpha release for later this year. We are actively seeking contributions from the community. To contribute [see the board here](https://github.com/google/android-fhir/projects/6)

### Structured data capture:

> The goal of this library is to implement the FHIR SDC specification on Android. A common use case that the SDC Library supports is the ability to translate a FHIR Questionnaire into a data capture fragment which can capture user input (via form controls) generating a QuestionnaireResponse resource. The current release (alpha-0.1.0) provides support for 10 commonly used widgets and very basic skip logic (see release note). This library will be incrementally updated and we are actively seeking contributions from the community. To contribute [see the board here](https://github.com/google/android-fhir/projects/1)

Author: Makes sense. I've edited that section. PTAL

> **NOTE:** The SDC library currently only supports expanded and inlined ValueSets (i.e. use of explicit answerOptions). See the demo application for samples.

Author: Have re-organised

## Sample Applications
Two sample applications are provided that demonstrate different features of the SDK. 

**These applications are provided for demonstration purposes only and are not for use in production. Do NOT use in production.**

### Clinical Data Repository (CDR) Application
This is the initial demo application for showcasing core SDK features. The demo currently uses synthea generated data that has been loaded into the HAPI fhir server public demo site.  On-loading the application a set of synthetic patient data is downloaded from the HAPI FHIR server to the local SQLLite DB.

- 1-way sync (from server to local DB)
- Show list of patients
- Search for a patient
- Show patient details

> **Future demo features include:**

Reviewer: Can we remove future roadmap items from the readme?

# Android FHIR SDK (early access) ![master](https://github.com/google/android-fhir/workflows/CI/badge.svg?branch=master)

The Android FHIR SDK is an Android library for managing FHIR resources on Android. The SDK enables developers to develop offline, mobile first healthcare applications that adhere to the [FHIR specification](https://www.hl7.org/fhir/). The SDK is being designed to support a number of common mobile health use cases including:

- FHIR Clinical Data Repository (manage FHIR resources natively and syncing with FHIR server)
- FHIR native data capture for case reporting (structured data capture specification)
- Client tracking and care coordination applications (custom applications leveraging the FHIR SDK back-end)
- Native support for on-device clinical decision support (via use of CQL)

> - Ability to run certain FHIR operations for generation of Care Plans and Indicators (for example)

Author: Don't want to speak specifically about managing FHIR resources? Or use as a clinical data repository?

- Data Access API (FHIR resource management using FHIR protos)
- Storage API (via Sqlite DB)
- Sync API (sync capabilities to a FHIR server)
- Search API
- Data Capture API (Structured Data Capture library)
- CQL Engine (using https://github.com/DBCG/cql_engine)

## Details of the repository
The repository is organised into two main libraries, *core* and *structured data capture* as well as sample demo applications for each (see sample applications).

### Core library:
This is the main SDK library providing core functionality required for use cases that require a mobile based clinical data repository using FHIR. This is currently in active development with a planned alpha release for later this year. We are actively seeking contributions from the community. To contribute [see the board here](https://github.com/google/android-fhir/projects/6)

### Structured data capture:
The goal of this library is to implement the FHIR SDC specification on Android. A common use case that the SDC Library supports is the ability to translate a FHIR Questionnaire into a data capture fragment which can capture user input (via form controls) generating a QuestionnaireResponse resource. 

> 

Author: Let's add this

> 

Author: Added

> The current release (alpha-0.1.0) provides support for **10 commonly used widgets and very basic skip logic** (see release note). This library will be incrementally updated and we are actively seeking contributions from the community. To contribute [see the board here](https://github.com/google/android-fhir/projects/1)

Author: Done.

> **NOTE:** The SDC library currently only supports expanded and inlined ValueSets (i.e. use of explicit answerOptions). See the [sample applications](#sample applications) for samples.

Author: Ok sure. I thought it might be worth being explicit at this point but agree. Next step is to start to build out more detailed wiki etc...

> **NOTE:** The SDC library currently only supports expanded and inlined ValueSets (i.e. use of explicit answerOptions). See the [sample applications](#sample applications) for samples.

Author: Pulled this out

## Sample Applications
Two sample applications are provided that demonstrate different features of the SDK. 

**These applications are provided for demonstration purposes only and are not for use in production. Do NOT use in production.**

### Clinical Data Repository (CDR) Application
This is the initial demo application for showcasing core SDK features. The demo currently uses synthea generated data that has been loaded into the HAPI fhir server public demo site.  On-loading the application a set of synthetic patient data is downloaded from the HAPI FHIR server to the local SQLLite DB.

- 1-way sync (from server to local DB)
- Show list of patients
- Search for a patient
- Show patient details

**Future demo features include:**
- New patient registration
- Edit patient profile information
- Creation of encounters and observations for a patient
- Bi-directional sync

> #### CQL Proof of Concept

Author: That's ok. Once we have a sense of this we can clean it up

Within the CDR demo is an example of performing a CQL operation. This was developed as part of a hackathon together with Bryn Rhodes (alphora) and utilises the cqf-ruler engine. [TODO: Explanation required]

### SDC Gallery App

> There is an SDC Gallery App with existing demo with a number of sample FHIR Questionnaires pre-loaded. 

Reviewer 2: ```suggestion
This app includes a number of pre-loaded sample FHIR Questionnaires. 
```

> The Gallery App showcases the structured data capture library functionality and the ability to easily deploy a FHIR Questionnaire and turn it into a data capture form. For supported widgets, features and known bugs, see the latest release notes.

Reviewer 2: ```suggestion
It uses the SDC library to create data capture forms using FHIR questionnaires. For supported widgets, features and known bugs, see the latest release notes.
```

> Contributing

Reviewer 2: remove this (seems like a duplicate)

Author: Ok. Will review

## Contributing

> The Android FHIR SDK is being developed together with a consortium of applicaiton developers working in Global Digital Health. We welcome contributions from anyone.

Reviewer 3: typo in word `application`

> The Android FHIR SDK is being developed together with a consortium of applicaiton developers working in Global Digital Health. We welcome contributions from anyone.

Reviewer 3: ```suggestion
The Android FHIR SDK is being developed by a consortium of application developers. We welcome contributions from anyone.
```

Author: Thank you. Resolved

The Android FHIR SDK (the SDK) is an Android library for building offline-capable, mobile-first healthcare applications using [FHIR](https://www.hl7.org/fhir/) resources on Android. It supports a number of common mobile health use cases including:

- FHIR Clinical Data Repository (manage FHIR resources natively and syncing with FHIR server)
- FHIR native data capture for case reporting (structured data capture specification)
- Client tracking and care coordination applications (custom applications leveraging the FHIR SDK back-end)
- Native support for on-device clinical decision support (via use of CQL)

> - Ability to run certain FHIR operations for generation of Care Plans and Indicators (for example)

Author: Reconsider this change
